File: tests/testcases/test_result.py
# Assume that student_submission module is imported and can be used freely
# Also assume that PASS(), RESULT(res), FAIL(), and NO_RESULT() already exist

def main():
    res = student_submission.numberSaver()
    if res == 54:
        PASS()
    elif res == 83:
        RESULT(83)
    else:
        FAIL()


main()

File: tests/testcases/test_result.c
// This is a made up example of a test case that demonstrates the functionality of the grader

int main() {
    int res = numberSaver();
    if (res == 54)
        return PASS;        // Macro defined in testcase.py at compilation step
                            // Equivalent to RESULT(100)
    else if (res == 83)
        return RESULT(83);  // Sets student score for the testcase to 83%
    else
        return FAIL;        // Equivalent to RESULT(0)
}